File: JieNote_backend-feature-export_note/app/api/v1/endpoints/note.py
from fastapi import APIRouter, HTTPException, Depends
from sqlalchemy.orm import Session
from app.schemas.note import NoteCreate, NoteUpdate, NoteFind
from app.utils.get_db import get_db
from app.curd.note import create_note_in_db, delete_note_in_db, update_note_in_db, find_notes_in_db

router = APIRouter()

@router.post("", response_model=dict)
def create_note(note: NoteCreate, db: Session = Depends(get_db)):
    new_note = create_note_in_db(note, db)
    return {"msg": "Note created successfully", "note_id": new_note.id}

@router.delete("/{note_id}", response_model=dict)
def delete_note(note_id: int, db: Session = Depends(get_db)):
    note = delete_note_in_db(note_id, db)
    if not note:
        raise HTTPException(status_code=404, detail="Note not found")
    return {"msg": "Note deleted successfully"}

@router.put("/{note_id}", response_model=dict)
def update_note(note_id: int, content: str, db: Session = Depends(get_db)):
    note = NoteUpdate(id=note_id, content=content)
    updated_note = update_note_in_db(note_id, note, db)
    if not updated_note:
        raise HTTPException(status_code=404, detail="Note not found")
    return {"msg": "Note updated successfully", "note_id": updated_note.id}

@router.get("", response_model=dict)
def get_notes(note_find: NoteFind = Depends(), db: Session = Depends(get_db)):
    notes, total_count = find_notes_in_db(note_find, db)
    return {
        "pagination": {
            "total_count": total_count,
            "page": note_find.page,
            "page_size": note_find.page_size
        },
        "notes": [note.model_dump() for note in notes]
    }

File: src/api/note.js
import http from '@/utils/http';

/**
 * 笔记相关 API
 */

/**
 * 获取笔记列表
 * @param {Object} params - 查询参数
 * @param {number} [params.id] - 笔记 ID
 * @param {number} [params.page] - 页码
 * @param {number} [params.page_size] - 每页大小
 * @param {number} [params.article_id] - 文献 ID
 * @param {number} [params.query] - 查询
 * @returns {Promise<any>} 笔记列表数据
 */
export const getNotes = (params = {}) => {
  return http.get('/notes/get', { params });
};

/**
 * 获取笔记标题列表
 * @param {Object} params - 查询参数
 * @param {number} [params.id] - 笔记 ID
 * @param {number} [params.page] - 页码
 * @param {number} [params.page_size] - 每页大小
 * @param {number} [params.article_id] - 文献 ID
 * @returns {Promise<any>} 笔记标题列表数据
 */
export const getNoteTitles = (params = {}) => {
  return http.get('/notes/title', { params });
};

/**
 * 创建笔记
 * @param {Object} data - 笔记数据
 * @param {number} data.article_id - 文献 ID
 * @param {string} data.content - 笔记内容
 * @param {string} data.title - 笔记标题
 * @returns {Promise<any>} 创建结果
 */
export const createNote = (data) => {
  return http.post('/notes/create', data);
};

/**
 * 更新笔记
 * @param {number|string} noteId - 笔记 ID
 * @param {Object} data - 更新数据
 * @param {string} [data.content] - 笔记内容
 * @param {string} [data.title] - 笔记标题
 * @returns {Promise<any>} 更新结果
 */
export const updateNote = (noteId, data) => {
  // 构建查询参数
  const params = {};
  if (data.content !== undefined) params.content = data.content;
  if (data.title !== undefined) params.title = data.title;
  
  return http.post(`/notes/${noteId}`, params);
};

/**
 * 删除笔记
 * @param {number|string} noteId - 笔记 ID
 * @returns {Promise<any>} 删除结果
 */
export const deleteNote = (noteId) => {
  return http.delete(`/notes/${noteId}`);
};

export default {
  getNotes,
  getNoteTitles,
  createNote,
  updateNote,
  deleteNote
};
